src/AudioInterface.cpp

```diff
@@ -534,7 +535,10 @@ void AudioInterface::fromSampleToBitConversion
         std::memcpy(output+2, &tmp_u8, 1); // 8bits = 1 bytes
         break;
     case BIT32 :
-        std::memcpy(output, input, 4); // 32bit = 4 bytes
+        tmp_sample = *input;
+        // not necessary yet:
+        // tmp_sample = std::max(-1.0f, std::min(1.0f, tmp_sample));
```

Comment:
if uncommented, this line would throw a compiler error, because it's comparing `float`s with a `double` - could you either fix or remove? the problems with committing code that is commented out! I think this might have worked if `tmp_sample` was a `sample_t`, but you've changed it to a `double`.

Reply:
changed 1.0f to 1.0

Reply:
Am I reading this thread correctly?  Looks like this line of commented code is the only point of discussion at this stage.

Reply:
I would prefer to revert to the onliner for 32bit but as godbolt told me that it's actually the same with -O2 and this already took way too much time, we should push this now.
[Godbolt Comparison](https://godbolt.org/z/nEhhv3)